src/core/ddsc/include/dds/dds.h

```diff
@@ -2668,6 +2668,42 @@ dds_take_mask_wl(
   uint32_t maxs,
   uint32_t mask);
 
+/**
+ * @brief Access the collection of CDR serialized data values (of same type) and
+ *        sample info from the data reader, readcondition or querycondition.
+ *
+ * This call accesses the data in CDR format from the data reader or readcondition
+ * and makes it available to the application. CDR is the Common Data Representation
+ * format that is used as the default ddsi wire format to exchange data between DDS
+ * participants. Once read the data is removed from the reader and cannot be 'read'
+ * or 'taken' again.
+ *
+ * Return value provides information about number of samples read, which will
+ * be <= maxs. Based on the count, the buffer will contain data to be read only
+ * when valid_data bit in sample info structure is set.
+ * The buffer required for data values, could be allocated explicitly or can
+ * use the memory from data reader to prevent copy. In the latter case, buffer and
+ * sample_info should be returned back, once it is no longer using the data.
+ *
+ * @param[in]  reader_or_condition Reader, readcondition or querycondition entity.
+ * @param[out] buf An array of pointers to samples in CDR format into which data is read (pointers can be NULL).
```

Comment:
As far as I can tell, ```struct ddsi_serdata **buf``` is not an array of pointers, but a pointer to an array of ```ddsi_serdata``` structures in which I don't spot any pointers to actual cdr data. Am I missing something?

Reply:
@martinbremmer You are correct, the suggestion that this is restricted to cdr data is incorrect.
The `buf` is a pointer to an array containing pointers to `struct ddsi_serdata`. This structure contains (references to) serialised data and functions to access the data, but not necessarily in cdr format.  So I agree with your comment and will change the description. Unfortunately I cannot change the rename of the API function called `take_cdr` because that would potentially break backward compatibility.